### PR TITLE
Backport spirv serialization and CI to 0.4 release

### PR DESCRIPTION
**Pull request by Frizi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,27 +9,35 @@ rust: stable
...
+    # # macOS 64bit
+    # - env: MACOSX_DEPLOYMENT_TARGET=10.9
+    #   os: osx
+    #   osx_image: xcode10.2
+    #   # We need travis_wait here to ensure the build doesn't timeout due to
+    #   # shader compilation
+    #   script: travis_wait 30 make
 
     # iPhoneOS 64bit
     - env: TARGET=aarch64-apple-ios
```
Comment (omni-viral) [inline .travis.yml]: Did you left only iPhone build intentionally?

File: shader/src/lib.rs [hunk 2]
```diff
@@ -198,63 +230,90 @@ impl ShaderSetBuilder {
...
+                    factory,
+                )?,
             );
         }
 
         if let Some(shader) = self.fragment.clone() {
             set.shaders.insert(
                 ShaderStageFlags::FRAGMENT,
-                create_storage(ShaderStageFlags::FRAGMENT, (shader.0, shader.1, spec_constants.fragment), factory)?,
+                create_storage(
```
Comment (omni-viral) [inline shader/src/lib.rs]: Looks weird. Would you kindly check that there are no trailing spaces?

File: .travis.yml (hunk 1 above)
Comment (Frizi) [reply .travis.yml]: Yes. See #209 for details. This is just a cherry-pick.

**Comment by Frizi** [comment]
This requires not only a review and merge, but also a `0.4.1` release after that's done.

**Comment by Frizi** [comment]
bors r=jaynus

**Comment by Frizi** [comment]
bors retry

**Comment by jaynus** [comment]
bors retry

**Pull request merged**